workflow-testing/src/main/java/com/squareup/workflow/testing/RenderIdempotencyChecker.kt

```diff
@@ -41,12 +41,17 @@ object RenderIdempotencyChecker : WorkflowInterceptor {
     proceed: (P, S, RenderContext<S, O>) -> R,
     session: WorkflowSession
   ): R {
-    val actualContext = RecordingRenderContext(context)
-        .also { recordingContext ->
-          proceed(props, state, recordingContext)
-          recordingContext.startReplaying()
+    val recordingContext = RecordingRenderContext(context)
+    proceed(props, state, recordingContext)
+
+    // The second render pass should not actually invoke any real behavior.
+    recordingContext.startReplaying()
+    return proceed(props, state, recordingContext)
+        .also {
+          // After the verification render pass, any calls to the context _should_ be passed
+          // through, to allow the real context to run its usual post-render behavior.
+          recordingContext.stopReplaying()
```

Comment:
So hacky.

Reply:
Yes, and now all the hacks are entirely contained in this one class!